pyomo/contrib/pyros/tests/test_grcs.py

```diff
@@ -271,6 +273,83 @@ def test_bounds_to_constraints(self):
+        m = ConcreteModel()
+        # parameters
+        m.p = Param(initialize=8, mutable=True)
+        m.r = Param(initialize=-5, mutable=True)
+        m.q = Param(initialize=1, mutable=False)
+        m.s = Param(initialize=1, mutable=True)
+
+        # variables, with bounds contingent on params
+        m.u = Var(initialize=0, bounds=(0, m.p))
+        m.v = Var(initialize=1, bounds=(m.r, m.p))
+        m.w = Var(initialize=1, bounds=(None, None))
+        m.x = Var(initialize=1, bounds=(0, exp(-1*m.p / 8) * m.q * m.s))
+        m.y = Var(initialize=-1, bounds=(m.r * m.p, 0))
+        m.z = Var(initialize=1, bounds=(0, m.s))
+
+        # objective
+        m.obj = Objective(sense=maximize, expr=m.x**2 - m.y + m.z**2 + m.v)
+
+        # clone model
+        mod = m.clone()
+        uncertain_params = [mod.p, mod.r]
+
+        # check variable replacement without performance constraints
+        replace_uncertain_bounds_with_constraints(mod, uncertain_params)
+        self.assertTrue(hasattr(mod, 'uncertain_var_bound_cons'),
+                        msg='Uncertain variable bounds erroneously added. '
+                            'Check only variables participating in active '
+                            'constraints are added.')
+        self.assertTrue(not mod.uncertain_var_bound_cons)
+
+        # constraints to add
+        constraints_m = ConstraintList()
+        m.add_component('perf_constraints', constraints_m)
+        constraints_m.add(m.w == 2 * m.x + m.y)
+        constraints_m.add(m.v + m.x + m.y >= 0)
+        constraints_m.add(m.y ** 2 + m.z >= 0)
+        constraints_m.add(m.x ** 2 + m.u <= 1)
+        constraints_m[4].deactivate()
+
+        # clone model with constraints added
+        mod_2 = m.clone()
+
+        # manually replace uncertain parameter bounds with explicit constraints
+        uncertain_cons = ConstraintList()
+        m.add_component('uncertain_var_bound_cons', uncertain_cons)
+        uncertain_cons.add(m.x - m.x._ub <= 0)
+        uncertain_cons.add(m.y._lb - m.y <= 0)
+        uncertain_cons.add(m.v - m.v._ub <= 0)
+        uncertain_cons.add(m.v._lb - m.v <= 0)
+        # remove corresponding variable bounds
+        m.x.setub(None)
+        m.y.setlb(None)
+        m.v.setlb(None)
+        m.v.setub(None)
+
+        # replace bounds in model with performance constraints
+        uncertain_params = [mod_2.p, mod_2.r]
+        replace_uncertain_bounds_with_constraints(mod_2, uncertain_params)
+
+        # check that vars participating in
+        # activated constraints correctly determined
+        svars = ComponentSet([v for v in get_vars_from_constraints(mod_2)])
+        vars_in_active_constraints = ComponentSet([mod_2.z, mod_2.w, mod_2.y,
+                                                   mod_2.x, mod_2.v])
+        self.assertEqual(svars, vars_in_active_constraints,
+                         msg='Mismatch in variables participating in '
+                             'activated constraints.')
```

Comment:
Do you check that the constraints you add are correct? You probably want to check that `mod_2.uncertain_var_bound_cons` contains the variables and parameters you expect it to. Right now it looks like you're just checking that model (which contains other constraints) contains the variables you expect it to.

Reply:
I agree with @Robbybp that this would be a good idea. I like to use `generate_standard_repn` on the constraint body for this--it's pretty easy since your constraints are linear. You could look at [these tests](https://github.com/Pyomo/pyomo/blob/main/pyomo/contrib/fme/tests/test_fourier_motzkin_elimination.py#L82) for a fairly self-contained example.